core/trino-main/src/main/java/io/trino/operator/aggregation/DecimalAverageAggregation.java

```diff
@@ -161,28 +161,31 @@ public static void inputLongDecimal(Type type, LongDecimalWithOverflowAndLongSta
 
     private static void accumulateValueInState(Slice unscaledValue, LongDecimalWithOverflowAndLongState state)
     {
-        accumulateAndUpdateOverflow(unscaledValue, state);
-        state.setLong(state.getLong() + 1);
-    }
-
-    private static void initializeIfNeeded(LongDecimalWithOverflowAndLongState state)
-    {
-        if (state.getLongDecimal() == null) {
-            state.setLongDecimal(UnscaledDecimal128Arithmetic.unscaledDecimal());
+        Slice sum = state.getLongDecimal();
+        if (sum == null) {
+            sum = UnscaledDecimal128Arithmetic.unscaledDecimal();
+            state.setLongDecimal(sum);
```

Comment:
you could skip `addWithOverflow` completely here as in `combine` method

Reply:
Unfortunately, no. The way this aggregation manages state is a little messy. It relies on side effects to update the state: 
* it sets the state to an empty slice if it's null
* if relies on addWithOverflow to update it in place

I think there are opportunities to clean this up and improve it later, but it will require a rewrite and specialization for short vs long decimals

Reply:
You could use `state.setLongDecimal(UnscaledDecimal128Arithmetic#unscaledDecimal(unscaledValue))` (it does copy slice)
Are we afraid that original `unscaledValue` will be modified?

Reply:
I didn't change the original logic (which I think is confusing and hard to reason about) -- I just restructured the code to avoid repeated calls to get state.

I'll leave that for a future revamp of this class.